Added Session Termination UseCase.

ujji8055 opened the pull request:
## Related Issues
Terminate User's SSO sessions.

## Description
Terminate all the user's SSO sessions on receiving **Account Lockout Alert**.

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

daryakoval commented on Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.py:
```diff
@@ -559,6 +560,168 @@ def validate_tenant_sta(self):
+            limit = self.validate_limit_sta(limit)
+            if limit <= 100:
+                query_params = (
+                    ('pageIndex', 0),
```
Is there an option to get to page number 3 using xsoar command?

ujji8055 replied:
No such parameter/argument for the commands to change the page no.

daryakoval commented on Packs/SafeNet_Trusted_Access/IncidentTypes/incidenttype-SafeNet_Trusted_Access_Account_Lockout_Alert.json:
```diff
@@ -0,0 +1,28 @@
+    "version": -1,
+    "weeks": 0,
+    "weeksR": 0,
+    "fromVersion": "5.5.0"
```
If there is no specific reason for this incident type to has fromvesion 5.5.0, please change to 6.0.0.
```suggestion
    "fromVersion": "6.0.0"
```

ujji8055 replied:
Done.

![image](https://user-images.githubusercontent.com/86044314/140712082-c2a03548-75bf-45c6-82df-101a5fa72f07.png)

daryakoval commented on Packs/SafeNet_Trusted_Access/Classifiers/classifier-SafeNet_Trusted_Access_–_Account_Lockout_Alert_Classifier.json:
```diff
@@ -0,0 +1,68 @@
+{
+    "description": "Classifies SafeNet Trusted Access account lockout alerts.",
+    "feed": false,
+    "id": "accccf10-97da-425d-840f-beac0233efb1",
```
```suggestion
    "id": "SafeNet Trusted Access – Account Lockout Alert Classifier",
```

ujji8055 replied:
Done. Updated ID to classifier name.

![image](https://user-images.githubusercontent.com/86044314/140712322-a1b8107e-3f44-44e5-8064-c0539a845607.png)

daryakoval commented on Packs/SafeNet_Trusted_Access/.pack-ignore:
```diff
@@ -1,5 +1,4 @@
-[file:SafeNet_Trusted_Access_–_Add_to_Unusual_Activity_Group.yml]
+[file:playbook-SafeNet_Trusted_Access_-_Add_to_Unusual_Activity_Group.yml]
+ignore=PB106
```
Why you are ignoring this error?

ujji8055 replied:
As discussed with Kobbi, this is affecting the usecase functionality. So as suggested, added this under the .pack-ignore file.
For our previous version pack, I am not able to see our playbook readme like https://xsoar.pan.dev/docs/reference/playbooks/abuse-inbox-management-detect--respond 
Please check and let us know the exact reason for the same.

kgal-pan replied:
This error indicates that the Playbook is bound to a specific Integration instance. I informed Joshi that this is not best practice (that the Integration instance name is used as an input to a Playbook). 

Joshi informed me that they have a guide to that instructs users to specify the instance name.

ujji8055 replied:
Thank you Kobbi.

ujji8055 commented on Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md:
```diff
@@ -0,0 +1,22 @@
+#### Integrations
+##### SafeNet Trusted Access
+- Added 5 commands.
```
Done. Updated as shown below.

![image](https://user-images.githubusercontent.com/86044314/140705610-aa6045e2-c6a0-4a99-a450-32020f845fe8.png)

ujji8055 commented on Packs/SafeNet_Trusted_Access/README.md:
```diff
@@ -3,6 +3,7 @@ SafeNet Trusted Access prevents data breaches, helps with compliance with regula
 
 -	Receive and process SafeNet Trusted Access alerts that indicate security risks to end user accounts.
--	Apply security remediation actions on SafeNet Trusted Access.
+-	Apply security remediation actions on SafeNet Truste
+d Access.
```
Done. Updated as shown below.

![image](https://user-images.githubusercontent.com/86044314/140705964-c919d470-a289-40f8-81f4-69a0e8926ead.png)

ujji8055 commented on Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.yml:
```diff
@@ -711,6 +711,216 @@ script:
       description: Checks if you have permission to access the requested tenant.
       type: boolean
+  - arguments:
+    - description: The maximum number of results to return.
+      name: limit
```
No, API doesn't have an offset parameter. 

![image](https://user-images.githubusercontent.com/86044314/140708718-046f6b67-1455-4d9d-b763-32849aa43a28.png)

ujji8055 commented on Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/README.md:
```diff
@@ -155,7 +155,7 @@ Get the profile information for a specific user.
 ### sta-create-user
 ***
-Create new user in the tenant
+Create a new user in the tenant.
```
Done. Updated for every occurance.

![image](https://user-images.githubusercontent.com/86044314/140709546-cb1cb1a9-f3c8-452b-ad30-c798104c91f3.png)

ujji8055 commented on Packs/SafeNet_Trusted_Access/Integrations/SafeNetTrustedAccess/SafeNetTrustedAccess.yml:
```diff
@@ -711,6 +711,216 @@ script:
+      name: userName
+      required: true
+    - description: The maximum number of results to return.
+      name: limit
```
By default (without limit parameter), it'll fetch information of the first page(index-0).
No default value for the limit. The maximum value is 10000. Validation is present in the code for the same.

ujji8055 commented on Packs/SafeNet_Trusted_Access/Playbooks/playbook-SafeNet_Trusted_Access_-_Add_to_Unusual_Activity_Group.yml:
```diff
@@ -1,35 +1,35 @@
 inputs:
 - description: Username of the user.
-  key: userName
+  key: UserName
```
This is suggested by the bussiness. As we have recently uploaded our previous versions and have less number of downloads. So it's ok.

daryakoval commented on Packs/SafeNet_Trusted_Access/ReleaseNotes/2_0_0.md:
```diff
@@ -0,0 +1,27 @@
+- This playbook terminates user's SSO sessions so that upon the next login attempt following the unlocking of the account, authentication is required. (Available from Cortex XSOAR 6.0.0).
+
+##### Updated: SafeNet Trusted Access - Add to Unusual Activity Group
+- This playbook adds the user to a group that was created to identify unusual activity. SafeNet Trusted Access policies can be configured to take this into account and provide stronger protection when handling access events from users who are members of the group. The user is added to this group for a configurable period of time.
```
```suggestion
- ***Breaking Change:*** Changed playbook inputs names. 
```

ujji8055 replied:
Done. As discussed, updated release note and added the **.json** file.

daryakoval reviewed (commented):
Please run `demisto-sdk update-release-notes -i Packs/<Pack Name> -bc`, 
to add breaking change note, do not forget to change `breakingChangesNotes` field in the new created json file, as explained [here](https://xsoar.pan.dev/docs/documentation/release-notes#breaking-changes-version).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=15589) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=15589) before we can accept your contribution.<br/>**7** out of **9** committers have signed the CLA.<br/><br/>:white_check_mark: ujji8055<br/>:white_check_mark: Itay4<br/>:white_check_mark: JasBeilin<br/>:white_check_mark: MosheEichler<br/>:white_check_mark: barchen1<br/>:white_check_mark: DeanArbel<br/>:white_check_mark: content-bot<br/>:x: edik24<br/>:x: agamm<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=15589) it.</sub>

daryakoval commented:
Hi, you can convert timestamp to datastring in the human readable of sta-get-user-sessions command, using `timestamp_to_datestring` [function](https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L6443). 
***sta-get-user-application*** - check why this command not working.

ujji8055 commented:
@daryakoval 
The command is working as expected. As discussed within the team, there was some issue with our test environment Swagger Rest API. Please find the screenshot below.

![image](https://user-images.githubusercontent.com/86044314/141258406-28a33048-4329-4c8d-af7d-c4d1ee0d6c63.png)

ujji8055 commented:
@daryakoval 
Please put the pack on hold for now. I am incorporating Elia comments. I will let you know once it's good to merge.

ujji8055 commented:
@evisochek - I have created a common classifier and mapper for both the use-case. As discussed with Kobbi, added Deprecated into old classifier files and updated the incident type name for every occurrence.
Kindly review and suggest the needful.
